qiskit/circuit/__init__.py

```diff
@@ -56,6 +56,50 @@
 Supplementary Information
 =========================
 
+.. dropdown:: Quantum Circuit with conditionals
+   :animate: fade-in-slide-down
+
+   When building a quantum circuit, there can be interest in applying a certain gate only
+   if a classical register has a specific value. This can be done with the
```

Comment:
You say "classical register" up here, but all the examples you use are actually a bit comparison internally - using an integer index is actually a bit reference.  For example, `c_if(0, 1)` is the same as `c_if(qc.clbits[0], True)`.  To get a register condition, you need to pass a register object directly.

It might be good to include an example both of the bit form (like you already have), _and_ the register form.

Reply:
It won't get seen if it's on that page, though `Instruction.c_if` would also be a valid place to put it, even if _technically_ that method is only called internally, and that might be seen.  I think you're fine to leave it on the circuit page, and perhaps to avoid repetition, just have one example that has two operations with conditions (one with bit, one with register)?

Reply:
`InstructionSet.c_if()` is the only `c_if()` that seems to have a complete reference page (I'm guessing to avoid duplication of content) so it wouldn't make much sense to put the examples in `Instruction.c_if()` unless we plan to move there all the content from `InstructionSet.c_if()`. 

So if we discard adding the code examples I guess our best option is putting together the two conditionals.

Reply:
Oh, sorry, I misread your comment - I thought you were saying that there's no page for `InstructionSet.c_if`.  Yeah, if it exists, you could put your examples there if you prefer.  I'm happy for you and Junye (@HuangJunye) to make the decision and put them wherever you think is sensible.

Reply:
I'd better go into the different inputs in the `InstructionSet.c_if` page, to which I'm already linking in this PR. My reasoning is that this section is mostly about getting users to know `InstructionSet.c_if` exists (so better to keep it simple) and the API is supposed to be covering the details (with code examples if they make sense)

Should I create a separate PR for the examples or it's better that I put them here?

Reply:
It's fine to put it in this PR if you want - it's logically the same change (to me, at least).